ms2rescore/peptideshaker.py

```diff
@@ -152,21 +287,101 @@ def _parse_modification(modified_seq):
 
         return mods_peprec
 
+    @staticmethod
+    def _calculate_top7_peak_features(
+        intensities: List,
+        mass_errors: List
+    ) -> Tuple[np.ndarray]:
+        """
+        Calculate "top 7 peak"-related search engine features.
+
+        The following features are calculated:
+        - mean_error_top7: Mean of mass errors of the seven fragment ion peaks with the
+          highest intensities
+        - sq_mean_error_top7: Squared MeanErrorTop7
+        - stdev_error_top7: Standard deviation of mass errors of the seven fragment ion
+          peaks with the highest intensities
+
+        """
+        if not (isinstance(intensities, list) and isinstance(mass_errors, list)):
+            return np.nan, np.nan, np.nan
+
+        else:
+            intensities = [float(i) for i in intensities]
+            mass_errors = [float(i) for i in mass_errors]
+
+            indices_most_intens = np.array(intensities).argsort()[-1:-8:-1]
+            mass_errors_top7 = [(mass_errors[i]) for i in indices_most_intens]
+            mean_error_top7 = np.mean(mass_errors_top7)
+            sq_mean_error_top7 = mean_error_top7 ** 2
+            stdev_error_top7 = np.std(mass_errors_top7)
+
+            return mean_error_top7, sq_mean_error_top7, stdev_error_top7
+
+    @staticmethod
+    def _calculate_ion_current_features(

```

Comment:
We should move this function to another module, to be shared with the maxquant pipeline, avoiding code duplication.